### DISCUSSION TITLE
useMutation does not read from ReactQueryConfigProvider

### DISCUSSION THREAD

**Discussion by dagda1** [TanStack/query#601] [General]
Hi,

Totally awesome package before I start.  Great work!

Am I correct in saying `useMutation` does not read from a `ReactQueryConfigProvider` that I have defined in the same way as `useQuery` does?

Is this by design or something that needs added or is there another reason?

**Comment by tannerlinsley** [comment]
For now, useMutation doesn't support automatic retries. You are more than welcome to do this, but generally automatic retries on mutations can lead to some interesting issues on the backend. You'll probably want to start using idempotency keys or something to dedupe mutation requests if you start doing automatic retries.

**Reply by dagda1** [reply]
thanks for the reply.  Do you have an example of using idempotency keys with mutations?  I think that might be a good idea in general

**Reply by tannerlinsley** [reply]
I don't have an example of that, but you could simply generate a unique id for the operation you are doing that won't change across retries, then send that in your json payload to your backend. Then on your backend, you need to keep track of if you have successfully applied a mutation and store its idempotency key and make sure you ignore further requests with that same key.

**Reply by tannerlinsley** [reply]
https://developer.squareup.com/blog/understanding-the-essentials-idempotency/
